tests/baselines/reference/generatorTypeCheck62.errors.txt

```diff
@@ -0,0 +1,67 @@
+generatorTypeCheck62.ts(24,62): error TS2345: Argument of type '(state: State) => Generator<never, State, unknown>' is not assignable to parameter of type '(a: State) => IterableIterator<State, void>'.
+  Call signature return types 'Generator<never, State, unknown>' and 'IterableIterator<State, void>' are incompatible.
+    The types returned by 'next(...)' are incompatible between these types.
+      Type 'IteratorResult<never, State>' is not assignable to type 'IteratorResult<State, void>'.
+        Type 'IteratorReturnResult<State>' is not assignable to type 'IteratorResult<State, void>'.
+          Type 'IteratorReturnResult<State>' is not assignable to type 'IteratorReturnResult<void>'.
+            Type 'State' is not assignable to type 'void'.
+generatorTypeCheck62.ts(32,62): error TS2345: Argument of type '(state: State) => Generator<any, State, unknown>' is not assignable to parameter of type '(a: any) => IterableIterator<any, void>'.
+  Call signature return types 'Generator<any, State, unknown>' and 'IterableIterator<any, void>' are incompatible.
+    The types returned by 'next(...)' are incompatible between these types.
+      Type 'IteratorResult<any, State>' is not assignable to type 'IteratorResult<any, void>'.
+        Type 'IteratorReturnResult<State>' is not assignable to type 'IteratorResult<any, void>'.
+          Type 'IteratorReturnResult<State>' is not assignable to type 'IteratorReturnResult<void>'.
+
+
+==== generatorTypeCheck62.ts (2 errors) ====
+    export interface StrategicState {
+        lastStrategyApplied?: string;
+    }
+    
+    export function strategy<T extends StrategicState>(stratName: string, gen: (a: T) => IterableIterator<T | undefined, void>): (a: T) => IterableIterator<T | undefined, void> {
+        return function*(state) {
+            for (const next of gen(state)) {
+                if (next) {
+                    next.lastStrategyApplied = stratName;
+                }
+                yield next;
+            }
+        }
+    }
+    
+    export interface Strategy<T> {
+        (a: T): IterableIterator<T | undefined, void>;
+    }
+    
+    export interface State extends StrategicState {
+        foo: number;
+    }
+    
+    export const Nothing1: Strategy<State> = strategy("Nothing", function*(state: State) {
+                                                                 ~~~~~~~~
+!!! error TS2345: Argument of type '(state: State) => Generator<never, State, unknown>' is not assignable to parameter of type '(a: State) => IterableIterator<State, void>'.
+!!! error TS2345:   Call signature return types 'Generator<never, State, unknown>' and 'IterableIterator<State, void>' are incompatible.

```

Comment:
Aside: We should special-case the error messages for comparing a generator/iterable/iterator - this is a really big stack just to say one of the return, next, or yield types don't match.